# Barlont Snakemake Workflow:

A Snakemake pipeline for de novo hybrid assembly and structural variant calling in barley using Oxford Nanopore and Illumina reads.

## Requirements


#### Conda =< 24.11.2

wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh

#### Snakemake =< 8.16.0

conda install -c bioconda -c conda-forge snakemake

## Getting started



## Running on SLURM

#### Install executor plugin for slurm (https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html)

pip install snakemake-executor-plugin-slurm



## Quick Tips










## Citation

Please cite out paper at: 


## Funding

This bioinformatic pipeline is published as part of the BarleyMicroBreed project, that has received funding from the European Union's Horizon Europe research and innovation programme under Grant Agreement No. 101060057.
